pkg/kudoctl/cmd/diagnostics/resource_funcs.go

```diff
@@ -0,0 +1,165 @@
+
+	kc, err := kudo.NewClient(s.KubeConfig, s.RequestTimeout, s.Validate)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create kudo client: %v", err)
+	}
+	c, err := kube.GetKubeClient(s.KubeConfig)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create kube client: %v", err)
+	}
+	instance, err := kc.GetInstance(opts.Instance, s.Namespace)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get instance %s/%s: %v", s.Namespace, opts.Instance, err)
+	}
+	if instance == nil {
+		return nil, fmt.Errorf("instance %s/%s not found", s.Namespace, opts.Instance)
+	}
+	return &ResourceFuncsConfig{
+		c:           c,
+		kc:          kc,
+		ns:          s.Namespace,
+		instanceObj: instance,
+		opts:        metav1.ListOptions{LabelSelector: fmt.Sprintf("%s=%s", kudoutil.OperatorLabel, instance.Labels[kudoutil.OperatorLabel])},
+		logOpts:     corev1.PodLogOptions{SinceSeconds: &opts.LogSince},
+	}, nil
+}
+
+// NewKudoResources is a configuration for Kudo controller related resources
+// panics if used to load Kudo CRDs (e.g. Instance etc.)
+func NewKudoResources(s *env.Settings) (*ResourceFuncsConfig, error) {
+	c, err := kube.GetKubeClient(s.KubeConfig)
+	if err != nil {
+		return nil, err
+	}
+	opts := metav1.ListOptions{LabelSelector: fmt.Sprintf("app=%s", appKudoManager)}
+	ns, err := c.KubeClient.CoreV1().Namespaces().List(opts)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get kudo system namespace: %v", err)
+	}
+	if ns == nil || len(ns.Items) == 0 {
+		return nil, fmt.Errorf("kudo system namespace not found")
+	}
+	return &ResourceFuncsConfig{
+		c:       c,
+		ns:      ns.Items[0].Name,
+		opts:    opts,
+		logOpts: corev1.PodLogOptions{},
+	}, nil
+}
+
+// Object implements runtime.Object and
+// metav1.Object interfaces.
+// copied from K8 internal type metaRuntimeInterface
+type Object interface {
+	runtime.Object
+	metav1.Object
+}
+
+func isKudoCR(o runtime.Object) bool {
+	kind := o.GetObjectKind().GroupVersionKind().Kind
+	return kind == "Instance" || kind == "Operator" || kind == "OperatorVersion"
+}
+
+type stringGetter func() string
+
+func (r *ResourceFuncsConfig) Instance() (runtime.Object, error) {
+	return r.instanceObj, nil
+}
+
+func (r *ResourceFuncsConfig) OperatorVersion(name stringGetter) func() (runtime.Object, error) {
+	return func() (runtime.Object, error) {
+		return r.kc.GetOperatorVersion(name(), r.ns)
+	}
+}
+
+func (r *ResourceFuncsConfig) Operator(name stringGetter) func() (runtime.Object, error) {
+	return func() (runtime.Object, error) {
+		return r.kc.GetOperator(name(), r.ns)
+	}
+}
+
+func (r *ResourceFuncsConfig) Deployments() (runtime.Object, error) {
+	obj, err := r.c.KubeClient.AppsV1().Deployments(r.ns).List(r.opts)
+	return obj, err
+}
+
+func (r *ResourceFuncsConfig) Pods() (runtime.Object, error) {
+	obj, err := r.c.KubeClient.CoreV1().Pods(r.ns).List(r.opts)
+	//return obj, err
+	_, _ = obj, err
```

Comment:
A debug remnant?

Reply:
yep. simulated an error return and forgot to remove it.